go/ephemeral/kex2_test.go

```diff
@@ -0,0 +1,161 @@
+package ephemeral
```

Comment:
note: this is the same test as `engine/kex2_test.go` with some added assertions for EKs